### PR TITLE
Add documentation for GRANT and REVOKE

### PR DESCRIPTION
**Pull request by amrutagokhale**
@petroav can you please take a look?

A note about grant/revoke limitation: At present, the grant and revoke commands are only supported for hive connector. There are two places where I can mention this limitation: 
(1) On the “Grant” and “Revoke” pages, which will describe the respective syntax, and will go under “SQL Statement Syntax” in here https://prestodb.io/docs/current/sql.html 
(2) On individual connector pages, there is a section called "<Connector Name> Limitations”. For ex., for MySQL connector, if you scroll all the way to the bottom on this page https://prestodb.io/docs/current/connector/mysql.html, there is a section called "MySQL Connector Limitations” which lists the SQL statements not yet supported for MySQL. I can similarly list grant/revoke as a limitation for every connector except hive. 
I am leaning towards approach (1), since I think, not explicitly mentioning it on the syntax page would be misleading. But the current Presto documentation seems to follow (2).

Let me know what you think.

### CODE REVIEW

**Comment by amrutagokhale** [comment]
@brian-rickman review please.

**Comment by mattsfuller** [comment]
Also add docs here about how to enable this feature.
https://prestodb.io/docs/current/connector/hive.html
https://prestodb.io/docs/current/connector/hive-security.html

File: presto-docs/src/main/sphinx/sql/grant.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    [ WITH GRANT OPTION ]
+
+Description
+-----------
+
+Grants the specified privileges to the specified grantee.
+
+Specifying ``ALL PRIVILEGES`` grants ``DELETE, INSERT`` and ``SELECT`` privileges.
+
+Specifying ``PUBLIC`` grants privileges to all grantees.
```
Comment (brian-rickman) [inline presto-docs/src/main/sphinx/sql/grant.rst]: s/all grantees/all users/

File: presto-docs/src/main/sphinx/sql/grant.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+The optional ``WITH GRANT OPTION`` clause grants the privileges to grant the specified privileges.
```
Comment (brian-rickman) [inline presto-docs/src/main/sphinx/sql/grant.rst]: The optional `WITH GRANT OPTION` clause allows the grantee to grant these same privileges to other users.

File: presto-docs/src/main/sphinx/sql/grant.rst [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+Examples
+--------
+
+Grant ``INSERT`` and ``SELECT`` privileges on the table ``orders`` to user ``alice``::
+
+    GRANT INSERT, SELECT
+    ON orders TO alice;
+
+Grant ``SELECT`` privilege on the table ``nation`` to all the users, additionally granting the privilege to grant ``SELECT`` privilege::
```
Comment (brian-rickman) [inline presto-docs/src/main/sphinx/sql/grant.rst]: s/to all the users/to all users/

This example doesn't make much sense.  If you give everyone SELECT privileges on a table, why would anyone need the ability to give someone else that same ability?

Also, you used `nation` in the description, and `orders` in the example.

File: presto-docs/src/main/sphinx/sql/grant.rst [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    GRANT SELECT
+    ON orders TO PUBLIC
+    WITH GRANT OPTION;
```
Comment (brian-rickman) [inline presto-docs/src/main/sphinx/sql/grant.rst]: How about:

Grant `SELECT` privilege on the table `nation` to Alice, additionally allowing Alice to grant `SELECT` privilege to other users::

GRANT SELECT 
ON nation to alice 
WITH GRANT OPTION;

Grant `SELECT` privilege on the table `orders` to everyone::

GRANT SELECT 
ON orders to PUBLIC;

File: presto-docs/src/main/sphinx/sql/revoke.rst [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+======
+
+Synopsis
+--------
+
+.. code-block:: none
+
+    REVOKE [ GRANT OPTION FOR ]
+    (privilege [, ...] | ALL PRIVILEGES)
+    ON TABLE? table_name FROM grantee
```
Comment (brian-rickman) [inline presto-docs/src/main/sphinx/sql/revoke.rst]: Is the question mark after TABLE intentional?

File: presto-docs/src/main/sphinx/sql/revoke.rst [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+The optional ``GRANT OPTION FOR`` clause also revokes the privileges to grant the specified privileges.
+
+Grant command succeeds provided the grantor possesses the specified privileges as well as the `GRANT OPTION` for those privileges.
+
+Examples
+--------
+
+Revoke ``INSERT`` and ``SELECT`` privileges on the table ``orders`` from user ``alice``::
+
+    GRANT
```
Comment (brian-rickman) [inline presto-docs/src/main/sphinx/sql/revoke.rst]: s/GRANT/REVOKE/

File: presto-docs/src/main/sphinx/sql/revoke.rst [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    INSERT, SELECT
+    ON orders FROM alice;
+
+Revoke ``SELECT`` privilege on the table ``nation`` from all the users, additionally revoking the privilege to grant ``SELECT`` privilege::
```
Comment (brian-rickman) [inline presto-docs/src/main/sphinx/sql/revoke.rst]: /all the users/all users/

Comment (brian-rickman) [inline presto-docs/src/main/sphinx/sql/revoke.rst]: If Alice has SELECT and GRANT privileges on nation, and I want to revoke GRANT while still allowing her to access the table, do I need to execute two commands?

REVOKE GRANT OPTION FOR 
SELECT
ON nation FROM alice;

GRANT SELECT
ON nation TO alice;

**Comment by brian-rickman** [comment]
I agree that it is better to note the connector limitation on the GRANT/REVOKE pages rather than on the individual connector pages.

File: presto-docs/src/main/sphinx/sql/grant.rst (hunk 3 above)
Comment (amrutagokhale) [inline presto-docs/src/main/sphinx/sql/grant.rst]: Good catch and great point about the use of `WITH GRANT OPTION`. I am changing the example as you suggested.

File: presto-docs/src/main/sphinx/sql/revoke.rst (hunk 5 above)
Comment (amrutagokhale) [inline presto-docs/src/main/sphinx/sql/revoke.rst]: No, that should instead be `[ TABLE ]` indicating that it is optional word.

File: presto-docs/src/main/sphinx/sql/revoke.rst (hunk 7 above)
Comment (amrutagokhale) [inline presto-docs/src/main/sphinx/sql/revoke.rst]: Yes, that's right. You will have to execute the two commands.

**Comment by amrutagokhale** [comment]
Updated. Also added limitation about support for hive connector.

File: presto-docs/src/main/sphinx/sql/revoke.rst [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+======
+
+Synopsis
+--------
+
+.. code-block:: none
+
+    REVOKE [ GRANT OPTION FOR ]
+    (privilege [, ...] | ALL PRIVILEGES)
+    ON [ TABLE ] table_name FROM grantee
```
Comment (sanjay990) [inline presto-docs/src/main/sphinx/sql/revoke.rst]: FROM ( grantee | PUBLIC )

File: presto-docs/src/main/sphinx/connector/hive-security.rst [hunk 9]
```diff
@@ -2,6 +2,32 @@
...
+
+``read-only``                                      Operations that read data or metadata, such as ``SELECT``,
+                                                   are permitted, but none of the operations that write data or
+                                                   metadata, such as ``CREATE``, ``INSERT`` or ``DELETE``, are
+                                                   allowed.
+
+``sql-standard``                                   Users are permitted to perform the operations as long as
+                                                   they have the required privileges as per the SQL standard.
+                                                   In this mode, Presto enforces the authorization checks for
+                                                   queries based on the privileges defined in Hive metastore.
```
Comment (brian-rickman) [inline presto-docs/src/main/sphinx/connector/hive-security.rst:25]: Maybe put a link here to the GRANT/REVOKE docs?

File: presto-docs/src/main/sphinx/connector/hive-security.rst [hunk 10]
```diff
@@ -2,6 +2,32 @@
...
+================================================== ============================================================
+
+
+Authentication
+==============
 The default security configuration of the :doc:`/connector/hive` does not use
 authentication when connecting to a Hadoop cluster. All queries are executed as
 the user who runs the Presto process, regardless of which user who submits the
```
Comment (brian-rickman) [inline presto-docs/src/main/sphinx/connector/hive-security.rst]: s/user who submits/user submits/

File: presto-docs/src/main/sphinx/sql/grant.rst [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+Synopsis
+--------
+
+.. code-block:: none
+
+    GRANT ( privilege [, ...] | ( ALL PRIVILEGES ) )
+    ON [ TABLE ] table_name TO ( grantee | PUBLIC )
+    [ WITH GRANT OPTION ]
+
+Usage of the term ``grantee`` denotes both, users and roles.
```
Comment (brian-rickman) [inline presto-docs/src/main/sphinx/sql/grant.rst]: Don't need a comma after 'both'.

File: presto-docs/src/main/sphinx/sql/grant.rst [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+
+Grants the specified privileges to the specified grantee.
+
+Specifying ``ALL PRIVILEGES`` grants ``DELETE, INSERT`` and ``SELECT`` privileges.
+
+Specifying ``PUBLIC`` grants privileges to all grantees.
+
+The optional ``WITH GRANT OPTION`` clause allows the grantee to grant these same privileges to others.
+
+For ``Grant`` statement to succeed, you should possess the specified privileges as well as the ``GRANT OPTION`` for those privileges.
```
Comment (brian-rickman) [inline presto-docs/src/main/sphinx/sql/grant.rst]: s/you/user executing it/

File: presto-docs/src/main/sphinx/sql/revoke.rst [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+Usage of the term ``grantee`` denotes both, users and roles.
+
+Description
+-----------
+
+Revokes the specified privileges from the specified grantee.
+
+Specifying ``ALL PRIVILEGES`` revokes ``DELETE, INSERT`` and ``SELECT`` privileges.
+
+Specifying ``PUBLIC`` revokes privileges from all grantees.
```
Comment (brian-rickman) [inline presto-docs/src/main/sphinx/sql/revoke.rst:23]: Does "all grantees" include the user executing the command?

File: presto-docs/src/main/sphinx/sql/revoke.rst [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+
+The optional ``GRANT OPTION FOR`` clause also revokes the privileges to grant the specified privileges.
+
+For ``Revoke`` statement to succeed, you should possess the specified privileges as well as the ``GRANT OPTION`` for those privileges.
```
Comment (brian-rickman) [inline presto-docs/src/main/sphinx/sql/revoke.rst]: s/you/user executing it/

File: presto-docs/src/main/sphinx/connector/hive-security.rst (hunk 9 above)
Comment (amrutagokhale) [inline presto-docs/src/main/sphinx/connector/hive-security.rst:25]: Do you have a specific place in mind for the link?

File: presto-docs/src/main/sphinx/sql/revoke.rst (hunk 13 above)
Comment (amrutagokhale) [inline presto-docs/src/main/sphinx/sql/revoke.rst:23]: Yes.

File: presto-docs/src/main/sphinx/connector/hive-security.rst (hunk 9 above)
Comment (amrutagokhale) [inline presto-docs/src/main/sphinx/connector/hive-security.rst:25]: I added the following line:
"Hive connector now supports `GRANT` and `REVOKE` statements using which you
can grant and revoke privileges to users or roles defined in Hive metastore.
See :doc:`/sql/grant` and :doc:`/sql/revoke` for their usages."

**Comment by amrutagokhale** [comment]
Updated.

File: presto-docs/src/main/sphinx/connector/hive-security.rst (hunk 9 above)
Comment (brian-rickman) [inline presto-docs/src/main/sphinx/connector/hive-security.rst:25]: I think you could simplify this.  I would put the following note at the bottom of the sql-standard section:

To alter these privileges, use the :doc:`/sql/grant` and :doc:`/sql/revoke` commands.

Comment (amrutagokhale) [inline presto-docs/src/main/sphinx/connector/hive-security.rst:25]: done.

File: presto-docs/src/main/sphinx/sql/revoke.rst (hunk 13 above)
Comment (brian-rickman) [inline presto-docs/src/main/sphinx/sql/revoke.rst:23]: Seems a little extreme.  Is there an admin user who can reverse that, or would it be necessary to drop and re-create the table to restore access?  Either way, we should make this explicit in the documentation.

Comment (amrutagokhale) [inline presto-docs/src/main/sphinx/sql/revoke.rst:23]: I checked SQL standard, and revoking privileges from `PUBLIC` role is listed. The `REVOKE` code in Presto invokes the underlying hive metastore API method, so the behavior is dependent on that implementation.

**Comment by brian-rickman** [comment]
LGTM

**Pull request closed without merging**